unistd/dir.c

```diff
@@ -440,6 +441,60 @@ DIR *opendir(const char *dirname)
 }
 
 
+DIR *fdopendir(int filedes)
+{
+	DIR *s = calloc(1, sizeof(DIR));
+	struct stat statbuf;
+
+	if (s == NULL) {
+		errno = ENOMEM;
+		return NULL;
+	}
+
+	if (fstat(filedes, &statbuf) < 0) {
+		free(s);
+		return NULL; /* fstat sets EBADF */
```

Comment:
Does this check whether or not `filedes` is a "valid file descriptor open for reading."?

> The fdopendir() function shall fail if:
> [EBADF] The fd argument is not a valid file descriptor open for reading.